pkgs/build-support/rust/sysroot/default.nix

```diff
@@ -0,0 +1,35 @@
+{ stdenv, rust, rustPlatform, buildPackages }:
+
+{ shortTarget, originalCargoToml, target, RUSTFLAGS }:
+
+rustPlatform.buildRustPackage {
+  name = "custom-sysroot";
+
+  nativeBuildInputs = [
+    buildPackages.python3 buildPackages.python3.pkgs.toml
+  ];
+
+  unpackPhase = ''
+    python3 ${./cargo.py}
+  '';
+
+  RUSTC_SRC = rustPlatform.rustcSrc;
```

Comment:
Cool. so this works now without `./tools`?

Reply:
Hmm, what's `./tools`?

Reply:
Oh, no, it still needs `./tools` in rustSrc. I've pushed code that adds it back again. I'll look into the size of `./tools` so we can decide if we want to add it to the existing nixpkgs rustSrc

Reply:
See https://github.com/NixOS/nixpkgs/pull/95542#discussion_r504500893